### PR TITLE
LOG-1390: Refactor functional benchmark utility to run for a fixed duration

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* refactors the functional collector benchmarker to run as a timed duration
* Adds the ability to supply a config
* Provides graphs of memory and cpu usage

cc @alanconway @eranra @syedriko 

### Links
* https://issues.redhat.com/browse/LOG-1390

Dependent on:
* https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter/pull/285
* custom image of our fluentd or yet to be submitted OAL PR

### CODE REVIEW

File: internal/cmd/functional-benchmarker/runners/cluster/cluster_runner.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+const containerVolumeName = "container"
+
+type ClusterRunner struct {
+	framework *functional.FluentdFunctionalFramework
+	config.Options
+}
+
+func (r *ClusterRunner) Deploy() {
+	testclient := client.NewNamesapceClient()
```
Comment (syedriko) [inline internal/cmd/functional-benchmarker/runners/cluster/cluster_runner.go:27]: Namespace (spelling)

File: internal/cmd/functional-benchmarker/config/config.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  tag kubernetes.*
+  read_from_head "true"
+  @label @FLUENTDFORWARD
+  <parse>
+    @type regexp
+    expression /^(?<@timestamp>[^\s]+) (?<stream>stdout|stderr)( (?<logtag>.))? (?<log>.*)$/
+    time_format '%Y-%m-%dT%H:%M:%S.%N%:z'
+    keep_time_key true
+  </parse>
+</source>
```
Comment (syedriko) [inline internal/cmd/functional-benchmarker/config/config.go:25]: I don't think we can avoid the concat stage, for post-CRI-O reassembly

Comment (jcantrill) [reply internal/cmd/functional-benchmarker/config/config.go:25]: this is a baseline config currently unused

**Pull request merged**